modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -538,6 +538,45 @@ def reindex(self, axis, labels, **kwargs):
         return self.__constructor__(new_modin_frame)
 
     def reset_index(self, **kwargs):
+        if self._modin_frame._index_cache is None:
+
+            def _reset(df, *columns_idx, partition_idx):
```

Comment:
```suggestion
            def _reset(df, *columns, partition_idx=None):
```

Reply:
It's better to require `partition_idx` parameter to avoid errors.

Reply:
Why? If the parameter is no longer passed, then we will see it: `TypeError: _reset() missing 1 required keyword-only argument: 'partition_idx'`.

What problem are you trying to solve by making this parameter optional?

Reply:
If we start to pass `partition_idx` in to remote functions as part of *args at the core dataframe layer, that will break this code. Since currently we pass `partition_idx` in to remote functions as part of **kwargs, I would prefer to have the named argument in the function definition.

Reply:
I tried your suggestion on artificial examples. Don't you think that the first option has more expected behavior than the second option?

```python
# current definition of `_reset` function (example)
>>> def test(*args, idx):
...   print(args, "\n", idx)

>>> test(1,2,3, idx=2)
(1, 2, 3)
 2

>>> test(1,2,3, 2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: test() missing 1 required keyword-only argument: 'idx'

# your suggestion about definition of `_reset` function (example)
>>> def test(*args, idx=None):
...   print(args, "\n", idx)

>>> test(1,2,3, idx=2)
(1, 2, 3)
 2

>>> test(1,2,3, 2)
(1, 2, 3, 2)
 None
```

Reply:
We know in advance that the keyword arg can't be None for the kernel and can check it if the higher layers pass a wrong value.

```python
def test(*args, idx=None):
    if idx is None:
        raise Exception
    print(args, "\n", idx)
```

Reply:
ping

Reply:
Why write our own check if python can check it itself, at the time of the function call? (python will raise `TypeError: test() missing 1 required keyword-only argument: 'idx'` if we start to pass `idx` into remote functions as part of `*args`)

Reply:
In the case of `def test(*args, idx):` `idx` - Keyword-only argument. 

"Keyword-only arguments are not required to have a default value." from https://peps.python.org/pep-3102/#specification

Reply:
I don't like such optional assumptions in any specification at all. That relaxes users and gives them freedom to do what they want, which may lead to misunderstanding the code by other developers/users.

Reply:
What optional assumptions are you talking about? And how does it relax users that this parameter does not have a default value?

Reply:
> What optional assumptions are you talking about?

Exactly this -> "Keyword-only arguments are not required to have a default value.", i.e., what not required is optional.

> And how does it relax users that this parameter does not have a default value?

In this concrete case it requires THE developer to know the function declaration to work around the problem and pass the argument with no issue.

I wonder if everyone is okay with the current code? Maybe I am too meticulous.

Reply:
> Exactly this -> "Keyword-only arguments are not required to have a default value.", i.e., what not required is optional.

It says here that this kind of parameter does not need a default value in the function signature at all, however, when calling, the user **must** specify some value and only in the following format: `idx=some_value`.

Reply:
Wow, such a big discussion you've had here :D

Personally, I would like to leave it as it is. Given [this comment](https://github.com/modin-project/modin/pull/5547#discussion_r1087860867), it really seems more reasonable to have this parameter as a non-optional keyword argument.

Reply:
cc @mvashishtha, @vnlitvinov, I would like to listen your thoughts as well.

Reply:
I'm siding with **Anatoly** here, Python `TypeError` message on function misuse would be clear enough to tell the developer what to do with an error.

Come on, this feature is from Python 3.0 ("point zero"!), accepted in 2006. It's time to break mental compatibility with 2.7 😉